features/step_definitions/candidates/registrations/verify_your_email_steps.rb

```diff
@@ -19,7 +19,34 @@
   @session_token = FactoryBot.create(:candidate_session_token)
 end
 
+Given("I change device") do
+  Capybara.reset_session!
+end
+
+Given "A clear registration_store" do
+  # We need to do this so when we fetch out the session key in the next step
+  # it's always the key we expect as there's only going to be one
+  # this will probably break parralel tests
+  Redis.current.keys("test:registrations*").each(&Redis.current.method(:del))
+end
+
 Given("I follow the verify link in the confirmation email") do
+  # this will probably break parralel tests
+  session_key = Redis.current.keys('test:registrations:*').last
```

Comment:
As you'd mentioned, this is racy as hell _but_ we shouldn't hold up releasing the feature for the test. I think leave it @wip for now and I'll revisit it in the sprint.

General mulling below:

Problem: we only transfer the 'secret' via email - which don't actually get sent

Solutions:
1. Control the secret - something overridden for servertest (ie like DfE Sign In/Fake CRM behaviour in the cucumber features)
2. Find another means of passing the secret - the race is caused by multiple places setting the registration keys - 
a) one option would be something in servertest only which sets a known key in redis with the uuid shared via the sign in link (greatly reducing scope for race)
b) we come up with another means of exposing emails from one side to the other - eg - write the email content to redis keys? This could potentially be useful elsewhere